### PR TITLE
Adds proxy_url to v1.Endpoint. Fixes #1385

### LINKED ISSUE #1385: Support proxy_url in Endpoint
`v1.types.Endpoint` does not map the Prometheus **proxy_url** option.
I use custom proxy servers to transform non-prometheus endpoints into prometheus scrape-able feeds, so the proxy_url option is super useful. Will submit PR for this.

### PR DESCRIPTION
**Pull request by nickman**
proxy_url support in v1.Endpoint would be super helpful.

### CODE REVIEW

File: pkg/client/monitoring/v1/types.go [hunk 1]
```diff
@@ -367,6 +367,8 @@ type Endpoint struct {
 	BasicAuth *BasicAuth `json:"basicAuth,omitempty"`
 	// MetricRelabelConfigs to apply to samples before ingestion.
 	MetricRelabelConfigs []*RelabelConfig `json:"metricRelabelings,omitempty"`
+	//Optional ProxyURL
```
Comment (brancz) [inline pkg/client/monitoring/v1/types.go]: There is a space missing between `//` and the start of the comment, and this needs a more thorough description of what it does.

File: pkg/client/monitoring/v1/types.go [hunk 2]
```diff
@@ -367,7 +367,7 @@ type Endpoint struct {
 	BasicAuth *BasicAuth `json:"basicAuth,omitempty"`
 	// MetricRelabelConfigs to apply to samples before ingestion.
 	MetricRelabelConfigs []*RelabelConfig `json:"metricRelabelings,omitempty"`
-	//Optional ProxyURL
+	// Optional ProxyURL e.g. http://proxyserver:2195. Directs scrapes to proxy through this endpoint
```
Comment (brancz) [inline pkg/client/monitoring/v1/types.go]: End sentence with a period. For it to be marked optional in the documentation this needs to be a pointer to a string (*string). Then you can also skip the “Optional” start in the comment.

File: pkg/prometheus/promcfg.go [hunk 3]
```diff
@@ -233,6 +233,10 @@ func generateServiceMonitorConfig(version semver.Version, m *v1.ServiceMonitor,
 	if ep.Path != "" {
 		cfg = append(cfg, yaml.MapItem{Key: "metrics_path", Value: ep.Path})
 	}
+	if ep.ProxyURL != "" {
```
Comment (brancz) [inline pkg/prometheus/promcfg.go]: With a string pointer this will need to be changed to != nil.

**Comment by brancz** [comment]
The build is failing, because some parts of the docs have not been updated, you need to run `make generate` once more since there were modifications.

**Comment by nickman** [comment]
Got it. Fixed. Thanks again.

//Nicholas

**Comment by brancz** [comment]
Perfect, all green now, thanks for your effort!

**Pull request merged**